code/modules/power/supermatter/supermatter.dm

```diff
@@ -35,54 +47,103 @@
 	var/safe_alert = "Crystalline hyperstructure returning to safe operating levels."
 	var/warning_point = 50
 	var/warning_alert = "Danger! Crystal hyperstructure instability!"
-	var/emergency_point = 500
+	var/emergency_point = 400
 	var/emergency_alert = "CRYSTAL DELAMINATION IMMINENT."
-	var/explosion_point = 900
+	var/explosion_point = 600
 
 	var/emergency_issued = 0
 
 	var/explosion_power = 8
 
 	var/lastwarning = 0				// Time in 1/10th of seconds since the last sent warning
+	var/last_zap = 0				// Time in 1/10th of seconds since the last tesla zap
 	var/power = 0
 
 	var/oxygen = 0					// Moving this up here for easier debugging.
 
 	//Temporary values so that we can optimize this
 	//How much the bullets damage should be multiplied by when it is added to the internal variables
 	var/config_bullet_energy = 2
-	//How much of the power is left after processing is finished?
-//	var/config_power_reduction_per_tick = 0.5
 	//How much hallucination should it produce per unit of power?
 	var/config_hallucination_power = 0.1
 
+	var/debug = 0
+
+	var/disable_adminwarn = FALSE
+
+	var/aw_normal = FALSE
+	var/aw_notify = FALSE
+	var/aw_warning = FALSE
+	var/aw_danger = FALSE
+	var/aw_emerg = FALSE
+	var/aw_delam = FALSE
+
 	var/obj/item/device/radio/radio
 
 	//for logging
 	var/has_been_powered = 0
 	var/has_reached_emergency = 0
 
+/obj/machinery/power/supermatter_shard/crystal
+	name = "supermatter crystal"
+	desc = "A strangely translucent and iridescent crystal."
+	base_icon_state = "darkmatter"
+	icon_state = "darkmatter"
+	anchored = TRUE
+	warning_point = 200
+	emergency_point = 2000
+	explosion_point = 3600
+	gasefficency = 0.25
+	explosion_power = 24
+
+
 /obj/machinery/power/supermatter_shard/New()
 	. = ..()
 	poi_list |= src
+	atmos_machinery += src
 	radio = new(src)
 	radio.listening = 0
 	investigate_log("has been created.", "supermatter")
 
 
+/obj/machinery/power/supermatter_shard/proc/handle_admin_warnings()
+	if(disable_adminwarn)
+		return
+
+	// Generic checks, similar to checks done by supermatter monitor program.
+	aw_normal = status_adminwarn_check(SUPERMATTER_NORMAL, aw_normal, "INFO: Supermatter crystal has been energised.", FALSE)
```

Comment:
Oh, it might be good to include a link to hop to the atom with these messages, so admins can hop in to intervene if bad things are happening